python/nano/src/bigdl/nano/tf/optimizers/sparse_adam.py

```diff
@@ -0,0 +1,204 @@
+"""SparseAdam optimizer implementation."""
+
+import tensorflow.compat.v2 as tf
+from keras import backend_config
+from keras.optimizer_v2 import optimizer_v2
+from tensorflow.python.util.tf_export import keras_export
+import tensorflow
+
+
+class SparseAdam(tensorflow.keras.optimizers.Adam):
+    r"""Optimizer that implements the SparseAdam algorithm.

```

Comment:
let's briefly describe the algorithm here and use the comment format similar to other python in bigdl